frontend/packages/topology/src/behavior/useDndDrag.tsx

```diff
@@ -57,8 +60,13 @@ const getOperation = (operation: DragSpecOperation | undefined): string => {
   );
 };
 
-const hasOperation = (operation: DragSpecOperation | undefined): boolean => {
-  return !!(operation && (typeof operation === 'string' || Object.keys(operation).length > 0));
+const hasOperation = (
+  operation: DragSpecOperationType<DragOperationWithType> | undefined,
+): boolean => {
+  return !!(
+    operation &&
+    (typeof (operation as any).type === 'string' || Object.keys(operation).length > 0)
```

Comment:
Can we use a proper type? Or a guard type function.
```suggestion
    (typeof (operation as DragOperationWithType).type === 'string' || Object.keys(operation).length > 0)
```